kernel/host/xmicterm/host/xmicterm.cc

```diff
@@ -129,16 +144,73 @@ class file_logger {
     std::unordered_map<uint16_t, std::unique_ptr<std::ofstream>> files;
 };
 
+/// Wrapper to the micctrl application, that controls the xeon phi coprocessor
+class mic_ctrl {
+public:
+    mic_ctrl(int adapter_)
+        :adapter(adapter_) {
+      }
+
+
+    void boot() {
+      std::stringstream ss;
+      ss << "sudo sh -c \"echo \\\"boot:elf:`pwd`/boot64.elf\\\" > /sys/class/mic/mic"
```

Comment:
I thought, that it would be easier to open the sysfs state file while being root and keep the file descriptor beyond the privilege drop. The same effect as sudo+echo can be achieved by write(fd, cstr, len). As far as I can see, there is no additional privilege check in the mpss kernel module.